### PR TITLE
test: add tests for OTPInput

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
Adds web & native tests for OTPInput

Chained to PR: #731

### CODE REVIEW

File: packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+  it('should be focussed when autoFocus flag is passed', () => {
+    const label = 'Enter OTP';
+    // eslint-disable-next-line jsx-a11y/no-autofocus
+    const { getAllByLabelText } = renderWithTheme(<OTPInput label={label} autoFocus />);
+    const allInputs = getAllByLabelText(/character/);
+    expect(allInputs[0]).toHaveFocus();
+  });
+
+  it('should have hidden input for form submissions', () => {
```
Comment (divyanshu013) [inline packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx:27]: As a future enhancement we should add tests for form submissions too, will help us catch regressions https://github.com/razorpay/blade/issues/739

File: packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      expect(input).toBeValid();
+      expect(input).toBeEnabled();
+    });
+    // asserting inside a loop leads to an issue with multiple parallel runs ref: https://github.com/dequelabs/axe-core/issues/3426
+    await assertAccessible(allInputs[0]);
+    await assertAccessible(allInputs[1]);
+    await assertAccessible(allInputs[2]);
+    await assertAccessible(allInputs[3]);
+  });
+});
```
Comment (divyanshu013) [inline packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx:143]: Optional: you can add another test for testing the focus management of individual inputs with arrow / tab keys

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx (hunk 2 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx:186]: Done! Thanks for the catch. Missed adding these 🙈 

Also added tests for backspace and delete key behaviours

**Review by saurabhdaware — APPROVED** [review]
Looks good to me 🌮

File: packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx [hunk 3]
```diff
@@ -137,6 +138,28 @@ describe('<OTPInput />', () => {
...
+    const allInputs = getAllByLabelText(/character/);
+    await user.type(allInputs[0], otp);
+    await user.click(allInputs[5]);
+    expect(allInputs[5]).toHaveFocus();
+    await user.keyboard('{Backspace}');
+    expect(allInputs[5]).toHaveValue('');
+    expect(allInputs[4]).toHaveFocus();
+    await user.keyboard('{Delete}');
+    expect(allInputs[4]).toHaveValue('');
+    expect(allInputs[3]).toHaveFocus();
```
Comment (divyanshu013) [inline packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx:160]: Shouldn't <kbd>Del</kbd> key delete the character in front of cursor 🤔

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx:160]: The input element only has 1 character at a time so deleting or backspacing should remove the character within that input itself, right?

Comment (divyanshu013) [reply packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx:160]: I think with <kbd>Del</kbd> it should delete the character in the next input box if cursor is after current number, will need to check some existing otp inputs in the wild. Can do a followup PR if needed.

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx:160]: Yeah, lets discuss in the standup. The behaviour might be a bit weird if the next input is empty but the one after that has some value. Also how do we handle focus here? With backspace we move the focus.

**Pull request merged**